zio-config/src/main/scala/zio/config/actions/Read.scala

```diff
@@ -1,94 +1,86 @@
 package zio.config.actions
 
 import zio.config.ReadErrors
-import zio.config.{ ConfigDescriptor }
-import zio.config.{ ConfigReport, ConfigSource, Details }
-import zio.{ config, Ref, UIO, ZIO }
+import zio.config.ConfigDescriptor
+import zio.config.{ ConfigSource }
+import zio.{ config, ZIO }
 
 object Read {
   // Read
-  final def read[A](configuration: ConfigDescriptor[A]): ZIO[ConfigSource, ReadErrors, (ConfigReport, A)] = {
+  final def read[A](configuration: ConfigDescriptor[A]): ZIO[ConfigSource, ReadErrors, A] = {
     def loop[B](
       configuration: ConfigDescriptor[B],
-      report: Ref[ConfigReport],
       previousDescription: String
-    ): ZIO[ConfigSource, ReadErrors, (ConfigReport, B)] =
+    ): ZIO[ConfigSource, ReadErrors, B] =
       configuration match {
-        case ConfigDescriptor.Succeed(value) => report.get.map(t => (t, value))
+        case ConfigDescriptor.Succeed(v) => ZIO.access(_ => v)
 
         case ConfigDescriptor.Source(path, propertyType) =>
           for {
             value <- config
                       .getConfigValue(path)
                       .mapError(err => ReadErrors(err))
-            r <- report
-                  .update(_.addDetails(Details(path, value, previousDescription)))
+
             result <- ZIO.fromEither(
                        propertyType
                          .read(path, value)
-                         .fold(r => Left(ReadErrors(r)), e => Right((r, e)))
+                         .fold(r => Left(ReadErrors(r)), e => Right(e))
                      )
 
           } yield result
 
-        case ConfigDescriptor.MapEither(c, f, _) =>
-          loop(c, report, previousDescription).flatMap {
-            case (r, src) => ZIO.fromEither(f(src)).bimap(err => ReadErrors(err), res => (r, res))
+        case ConfigDescriptor.XmapEither(c, f, _) =>
+          loop(c, previousDescription).flatMap { a =>
+            ZIO.fromEither(f(a)).bimap(err => ReadErrors(err), res => res)
           }
 
+        // No need to add report on the default value.
+        case ConfigDescriptor.Default(c, value) =>
+          loop(c, previousDescription).fold(
+            _ => value,
```

Comment:
I think it would be better to inspect the error here. If the value is missing, the default value should be supplied. If there is a parse error or some fatal error, we should report the error (i.e., not supply a default value).

Reply:
Yes. I think it makes sense

Reply:
I will take this in follow up PR.